### PR TITLE
Revert changes to Superherofly

### PR DESCRIPTION
**Pull request by Tayipc38**
The changes done after the initial contribution broke more than half of the features in superherofly.

### CODE REVIEW

**Comment by iwanttodo1t** [comment]
I really liked the previous flying function, it was more fun, but this superhero mode feels no different from free movement.
Don't know why that was deleted.

File: src/backend/looped/self/super_hero_fly.cpp [hunk 1]
```diff
@@ -391,7 +399,7 @@ namespace big
 
 		virtual void on_enable() override
 		{
-			g_notification_service.push("SUPER_HERO_FLY"_T.data(), "SUPER_HERO_FLY_ENABLE_NOTIFICATION"_T.data());
+			g_notification_service.push("Super hero fly", "SUPER_HERO_FLY_ENABLE_NOTIFICATION"_T.data());
```
Comment (xiaoxiao921) [inline src/backend/looped/self/super_hero_fly.cpp]: No

**Review by xiaoxiao921 — CHANGES_REQUESTED** [review]

File: src/backend/looped/self/super_hero_fly.cpp (hunk 1 above)
Comment (Yimura) [reply src/backend/looped/self/super_hero_fly.cpp]: No

Comment (Tayipc38) [reply src/backend/looped/self/super_hero_fly.cpp]: Lol

**Comment by Tayipc38** [comment]
> I really liked the previous flying function, it was more fun, but this superhero mode feels no different from free movement. Don't know why that was deleted.

Could you try it again after this revert? I do not share your opinion.

**Review by Yimura — APPROVED** [review]

File: src/backend/looped/self/super_hero_fly.cpp [hunk 2]
```diff
@@ -22,7 +22,7 @@ namespace big
 
 		bool m_timer = false;
 		std::chrono::system_clock::time_point m_time_stamp;
-		std::chrono::milliseconds m_duration;
+		std::chrono::duration<double> m_duration;
```
Comment (gir489returns) [inline src/backend/looped/self/super_hero_fly.cpp:25]: std::chrono::milliseconds types as std::chrono::duration<long long>, why are you changing it to [double](https://en.wikipedia.org/wiki/Double-precision_floating-point_format), which is a double precision float?

File: src/backend/looped/self/super_hero_fly.cpp [hunk 3]
```diff
@@ -249,7 +257,7 @@ namespace big
 			PAD::DISABLE_CONTROL_ACTION(0, (int)ControllerInputs::INPUT_COVER, false);
 
 			if (m_timer)
-				m_duration = std::chrono::duration_cast<std::chrono::milliseconds> (std::chrono::system_clock::now() - m_time_stamp);
+				m_duration = std::chrono::system_clock::now() - m_time_stamp;
```
Comment (gir489returns) [inline src/backend/looped/self/super_hero_fly.cpp:260]: Why? If you're worried about accuracy being lost in calculations between calls, you can keep a std::chrono::high_resolution_clock type, and then set the m_duration to a std::chrono::duration_cast at the last moment.

File: src/backend/looped/self/super_hero_fly.cpp [hunk 4]
```diff
@@ -289,7 +297,7 @@ namespace big
 			}
 
 			if (PAD::IS_DISABLED_CONTROL_PRESSED(0, (int)ControllerInputs::INPUT_COVER) && !m_flying
-			    && g.self.super_hero_fly.charge)
+			    && g.self.super_hero_fly.charge && !PED::IS_PED_IN_ANY_VEHICLE(self::ped, false))
```
Comment (gir489returns) [inline src/backend/looped/self/super_hero_fly.cpp:300]: This change doesn't make any sense, you added a return check at the head of the function to see if you're in any vehicle, why are you checking it again?

**Review by gir489returns — CHANGES_REQUESTED** [review]

File: src/backend/looped/self/super_hero_fly.cpp (hunk 2 above)
Comment (Tayipc38) [reply src/backend/looped/self/super_hero_fly.cpp:25]: One of these changes fixes the features, just reverted it.

File: src/backend/looped/self/super_hero_fly.cpp (hunk 4 above)
Comment (Tayipc38) [reply src/backend/looped/self/super_hero_fly.cpp:300]: Funny thing, it still won't work lol

**Comment by Tayipc38** [comment]
Btw, I might try to make this a lua instead, sounds much more appropriate for a feature such as this.

**Comment by gir489returns** [comment]
> Btw, I might try to make this a lua instead, sounds much more appropriate for a feature such as this.

I'll take your word that this has been tested and fixed, although like you mentioned this might be better suited as a Lua script. Yim already approved the code, and the problems I have with it are lexical, however, given the fact this is a "fun" feature and not a core element, I think we can let code quality slide.

File: src/backend/looped/self/super_hero_fly.cpp (hunk 4 above)
Comment (iwanttodo1t) [reply src/backend/looped/self/super_hero_fly.cpp:300]: I use this to confirm that the player is inside the vehicle.
scr_globals::globalplayer_bd.as<GlobalPlayerBD*>()->Entries[self::id].PlayerBlip.NumPassengersInVehicle > 0

**Pull request merged**